File: sast-go/cases/completeness/single_app_tracing/exception_error/exception_throw/exception_throw_002_F/exception_throw_002_F.go
// evaluation information start
// real case = false
// evaluation item = 完整度->单应用跟踪完整度->异常与错误处理->异常抛出与捕获
// scene introduction = 异常抛出->函数内抛出
// level = 2
// bind_url = completeness/single_app_tracing/exception_error/exception_throw/exception_throw_002_F/exception_throw_002_F
// evaluation information end

package main

import "os/exec"

func exception_throw_002_F(__taint_src string) {
	defer func() {
		if r := recover(); r != nil {
			__taint_sink(r)
		}
	}()

	panic("_")

}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", o.(string)).Run()
}

func main() {
	__taint_src := "taint_src_value"
	exception_throw_002_F(__taint_src)
}

File: sast-js/case/accuracy/path_sensitive/exception_throw/exception_throw_002_F.js
// evaluation information start
// real case = false
// evaluation item = 准确度->路径敏感分析->异常抛出和捕获
// scene introduction = 异常抛出-try块
// level = 3
// bind_url = accuracy/path_sensitive/exception_throw/exception_throw_002_F
// evaluation information end

function exception_throw_002_F(__taint_src) {
  try {
    throw "_";
    __taint_sink(__taint_src);
  } catch (e) { }
}

function __taint_sink(o) { }
